src/layers/microscopy-viewer-layer/microscopy-viewer-layer.js

```diff
@@ -19,25 +21,37 @@ export class MicroscopyViewerLayer extends CompositeLayer {
   }
 
   updateState() {
-    if (!this.state.connections) {
+    /* eslint-disable no-bitwise */
+    if (
```

Comment:
Let me see if this can be more clear...

Reply:
Is what we're really interested in a discrepancy between state and props? (Also wondering about the necessity of having both state and props here, but set that to the side.) Would something like this work?:

```
if (this.props.useTiff && !this.state.isTiff) {
 ...
} else if (this.props.useZarr && !this.state.isZarr) {
 ...
}
```